docs/runbook.md

```diff
@@ -13,103 +13,317 @@
+  - name: name
+    value: Azure Test
+  - name: test_project
+    value: Test Project 1
+  - name: test_pass
+    value: Test Pass 1
+  - name: test_tag
+    value: round 1
+```
 
 ### notifier
 
+Receive messages during the test run and output them somewhere. For more info, please refer [Notifier](https://github.com/microsoft/lisa/blob/main/docs/extension.md#notifier).
+
 #### console
 
+This is the simplest notifier. It outputs messages to the console and file log and demonstrates how to implement notification procedures.
+
+##### log_level
+
+type: str, optional, default: DEBUG, values: DEBUG, INFO, WARNING...
+
+Set log level of notification messages.
+
 #### html
 
+Output test results in html format. It can be used for local development or as the body of an email.
+
+##### path
+
+type: str, optional, default: lisa.html
+
+Specify the output file name and path.
+
+##### auto_open
+
+type: bool, optional, default: False
+
+When set to True, the html will be opened in the browser after completion. Useful in local run.
+
+Example of notifier:
+```
+notifier:
+  - type: console
+    log_level: INFO
+  - type: html
+    path: .\lisa.html
+    auto_open: true
+```
 
 ### environment
 
-#### name
+List of environments. For more info, refer [node and environment](https://github.com/microsoft/lisa/blob/main/docs/concepts.md#node-and-environment).
+
+#### environments
+
+Define test run environment.
+
+##### name
+
+type: str, optional, default is empty
 
-#### topology
+Specify the name of the environment.
 
-#### nodes
+##### topology
 
-#### nodes_requirement
+type: str, optional, default is "subnet"
 
-##### type
+Specify the topology of the environment.
+
+##### nodes
+
+List of node, it can be a virtual machine on Azure or Hyper-V, bare metal or others. For more info, refer [node and environment](https://github.com/microsoft/lisa/blob/main/docs/concepts.md#node-and-environment).
+
+##### nodes_requirement
+
+Testing required environment, e.g. node_count (default is 1), core_count (default is 1), memory_mb (default is 512 MB), disk_count (default is 1) etc. The node can be created once the node requirement is met.
+
+###### type
+
+type: str, optional, default value is "requirement", supported values are "requirement", "remote", "local".
+
+Example of requirement environment:
+```
+environment:
+  environments:
+    - nodes:
+        - type: requirement
+          core_count:
+            min: 4
+          azure:
+            gallery: $(gallery_image)
+            location: $(location)
+            vm_size: $(vm_size)
+```
 
 ### platform
 
+List of platform, default value is "ready", current support values are "ready", "azure".
+
+Example of Azure platform:
+```
+platform:
+  - type: azure
+    admin_private_key_file: $(admin_private_key_file)
+    reserve_environment: $(reserve_environment)
+    azure:
+      subscription_id: $(subscription_id)
+      wait_delete: false
+```
+
 ### testcase
+
+type: list of str, optional, default: empty
```

Comment:
What's option of type? default is lisa, not empty. It means lisa native test cases.
It needs to explain the legacy parameters here.